### ISSUE TITLE
ctapipe-extra and caching issues in production

### ISSUE THREAD

**Issue by vuillaut** [cta-observatory/cta-lstchain#661]
Hi

This issue as been first reported by @garciagenrique there:
https://github.com/cta-observatory/ctapipe/issues/1668

However, this is not really a ctapipe issue, but rather lstchain and/or accounts configuration in La Palma.

The problem:
- lstchain uses ctapipe-extra to get some telescope information (such as focal length) through the `.from_name` mechanism
- with ctapipe 0.10, these info are downloaded in the cache in HOME 
- lstanalyzer accounts used for the production connects to the tcs06, where its HOME is
- the jobs then run on the computing nodes (cp**) - these don't have access to the lstanalyzer HOME leading to 
```PermissionError: [Errno 13] Permission denied: '/local'```

Ways to fix it:
- modify lstanalyzer configuration
- remove all references using ctapipe-extra in lstchain

@morcuended Have you tried running lstosa with lstchain v0.7? If so, I expect you run into the same issue... ?

**Comment by rlopezcoto** [comment]
> * remove all references using ctapipe-extra in lstchain

I am +1 for this solution. We can make a new bugfix release with this fix, #660 and maybe other.

**Comment by vuillaut** [comment]
> > * remove all references using ctapipe-extra in lstchain
> 
> I am +1 for this solution. We can make a new bugfix release with this fix, #660 and maybe other.

I agree.

**Comment by maxnoe** [comment]
> lstchain uses ctapipe-extra 

No it does not. It uses `ctapipe.utils.get_dataset_path`, which looks at multiple locations in this order:

1. directory pointed to by the env variable `CTAPIPE_SVC_PATH` if set
2. ctapipe-extra if it is installed
3. The download cache 

If it doesn't find the file in any of those, it tries to download from 

http://cccta-dataserver.in2p3.fr/data/ctapipe-extra/v0.3.3/

**Comment by morcuended** [comment]
The observed-data part of lstchain (R0 -> DL1 -> DL1AB -> DATACHECK -> DL2) works with the current master version without the need of exporting `CTAPIPE_SVC_PATH` with `lstanalyzer` account. In any case, LSTOSA always exports this service directory when running in the cluster.

**Comment by rlopezcoto** [comment]
> The observed-data part of lstchain (R0 -> DL1 -> DL1AB -> DATACHECK -> DL2) works with the current master version without the need of exporting `CTAPIPE_SVC_PATH` with `lstanalyzer` account. In any case, LSTOSA always exports this service directory when running in the cluster.

Also sending jobs with slurm? After #663, we still have `get_dataset_path` in several places, that will look for the data in `CTAPIPE_SVC_PATH` if it cannot download it, right? What did it change (`ctapipe-extra` installation, as point 2. in @maxnoe response https://github.com/cta-observatory/cta-lstchain/issues/661#issuecomment-809222953?)

**Comment by morcuended** [comment]
More notes on this. The tests I did with `lstanalyzer` account that worked fine were without setting `CTAPIPE_SVC_PATH` directory. It turns out that when trying to export the variable I did it wrong so lstchain tried to go to the next step, searching for the info in the `ctapipe-extra` package that was installed in the environment I'm using. That did not work so the last step downloading the info from the server to the local cache also failed.

**Comment by maxnoe** [comment]
> Also sending jobs with slurm? After #663, we still have get_dataset_path in several places, that will look for the data in CTAPIPE_SVC_PATH if it cannot download it, right?

No! As I wrote above, ctapipe **first** checks `CTAPIPE_SVC_PATH`. The download is the last resort.

**Comment by maxnoe** [comment]
> searching for the info in the ctapipe-extra package that was installed in the environment I'm using. That did not work

What do you mean with "did not work?" If ctapipe-extra is installed and all the files we need here should be in ctapipe extra, there should be no problem.

**Comment by morcuended** [comment]
Here I had `ctapipe-extra` installed and still got this error:
```
Traceback (most recent call last):
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/pathlib.py", line 1273, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/local/home/lstanalyzer/.cache/ctapipe/cccta-dataserver.in2p3.fr/data/ctapipe-extra'
...
Traceback (most recent call last):
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/bin/lstchain_data_r0_to_dl1", line 5, in <module>
    from lstchain.scripts.lstchain_data_r0_to_dl1 import main
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/lstchain/scripts/lstchain_data_r0_to_dl1.py", line 29, in <module>
    from lstchain.reco import r0_to_dl1
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/lstchain/reco/r0_to_dl1.py", line 175, in <module>
    input_filename=get_dataset_path('gamma_test_large.simtel.gz'),
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/ctapipe/utils/datasets.py", line 145, in get_dataset_path
    filepath = find_in_path(filename=filename, searchpath=searchpath)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/ctapipe/utils/datasets.py", line 115, in find_in_path
    for directory in get_searchpath_dirs(searchpath):
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/ctapipe/utils/datasets.py", line 41, in get_searchpath_dirs
    searchpaths.append(get_cache_path(DEFAULT_URL))
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/ctapipe/utils/download.py", line 75, in get_cache_path
    path.parent.mkdir(parents=True, exist_ok=True)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/pathlib.py", line 1277, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/pathlib.py", line 1277, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/pathlib.py", line 1277, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  [Previous line repeated 4 more times]
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/pathlib.py", line 1273, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '/local'
```

**Comment by maxnoe** [comment]
Ok, that's a ctapipe bug. It should catch that permission error.

**Comment by rlopezcoto** [comment]
but which permission error? According to https://github.com/cta-observatory/cta-lstchain/issues/661#issuecomment-809222953, the last thing it should do is to search for the files in the download cache. In the example @morcuended is reporting, `ctapipe-extra` is installed and it should search for the files there before than in the download cache, right?

**Comment by maxnoe** [comment]
Fix here: https://github.com/cta-observatory/ctapipe/pull/1670

In the mean time, the issue can be fixed by setting the env variable `CTAPIPE_CACHE` to a location writable by the lstanalyzer user.

**Comment by maxnoe** [comment]
This was mostly solved by this PR here: #864 and by downloading all data on the test data server to:

```
/fefs/aswg/data/ctapipe_service
```

You can prevent all downloads by setting the `CTAPIPE_SVC_PATH` environment variable like this:

```
export CTAPIPE_SVC_PATH=/fefs/aswg/data/ctapipe_service
```
